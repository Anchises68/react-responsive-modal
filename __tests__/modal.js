import React from 'react';
import { mount } from 'enzyme';
import Modal from '../src/modal';

const defaultProps = {
  classes: {},
  open: false,
  onClose: jest.fn()
}

describe('modal', () => {
  it('should have a default export', () => {
    expect(typeof Modal).toBe('function');
  });

  describe('render', () => {
    it('should render null when then modal is not opened', () => {
      const wrapper = mount(
        <Modal {...defaultProps}>
          <div>modal content</div>
        </Modal>
      );
  
      expect(wrapper.find(Modal).length).toBe(1);
      expect(wrapper.find(Modal)).toBeEmptyRender();
    });
  
    it('should render the content', () => {
      const wrapper = mount(
        <Modal {...defaultProps} open>
          <div>modal content</div>
        </Modal>
      );
  
      expect(wrapper.find(Modal).length).toBe(1);
      expect(wrapper.find(Modal)).toMatchSnapshot();
    });
  });
});
