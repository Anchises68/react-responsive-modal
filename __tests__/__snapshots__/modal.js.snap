// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal render should render the content 1`] = `
<Modal
  animationDuration={500}
  classNames={Object {}}
  classes={Object {}}
  closeIconSize={28}
  closeIconSvgPath={
    <path
      d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
    />
  }
  closeOnEsc={true}
  closeOnOverlayClick={true}
  little={false}
  onClose={[MockFunction]}
  open={true}
  showCloseIcon={true}
  styles={Object {}}
>
  <Portal>
    <CSSTransition
      appear={true}
      classNames={
        Object {
          "appear": undefined,
          "appearActive": undefined,
          "enter": undefined,
          "enterActive": undefined,
          "exit": undefined,
          "exitActive": undefined,
        }
      }
      in={true}
      onExited={[Function]}
      timeout={500}
    >
      <Transition
        appear={true}
        enter={true}
        exit={true}
        in={true}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={500}
        unmountOnExit={false}
      >
        <div
          className=""
          onMouseDown={[Function]}
        >
          <div
            className=""
          >
            <div>
              modal content
            </div>
            <CloseIcon
              classNames={Object {}}
              classes={Object {}}
              closeIconSize={28}
              closeIconSvgPath={
                <path
                  d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
                />
              }
              onClickCloseIcon={[Function]}
              styles={Object {}}
            >
              <button
                className=""
                onClick={[Function]}
              >
                <svg
                  className=""
                  height={28}
                  viewBox="0 0 36 36"
                  width={28}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
                  />
                </svg>
              </button>
            </CloseIcon>
          </div>
        </div>
      </Transition>
    </CSSTransition>
  </Portal>
</Modal>
`;
